### PR TITLE
Mutation testing, PHP 7.1 testing

### PR DESCRIPTION
**Pull request by Ocramius**
This PR enforces mutation testing on the package.

### CODE REVIEW

File: test/PackageVersionsTest/InstallerTest.php [hunk 1]
```diff
@@ -89,6 +89,7 @@ public function testDumpVersionsClass()
 
         $expectedPath = $vendorDir . '/ocramius/package-versions/src/PackageVersions';
 
+        /** @noinspection MkdirRaceConditionInspection */
```
Comment (kalessil) [inline test/PackageVersionsTest/InstallerTest.php]: @Ocramius : I'd just ignore test-classes in the inspection, bunch of inspection are doing ths already.

File: src/PackageVersions/Installer.php [hunk 2]
```diff
@@ -124,12 +124,11 @@ private static function writeVersionClassToFile(
...
-            self::locateRootPackageInstallPath($composerConfig, $rootPackage)
-            . '/src/PackageVersions/Versions.php',
-            $versionClassSource,
-            0664
-        );
+        $installPath = self::locateRootPackageInstallPath($composerConfig, $rootPackage)
+            . '/src/PackageVersions/Versions.php';
+
+        file_put_contents($installPath, $versionClassSource);
+        chmod($installPath, 0664);
```
Comment (kalessil) [inline src/PackageVersions/Installer.php]: @Ocramius : perhaps we should have an inspection for this?

Comment (Ocramius) [inline src/PackageVersions/Installer.php]: A default `umask()` will usually be `0022` or such, so it is generally OK to use `file_put_contents()`, but this file is a generated PHP file, and therefore vulnerable to code injection if the `umask()` is incorrect.

Interestingly, specifying the mask on `file_put_contents()` doesn't behave as expected :S

Comment (kalessil) [inline src/PackageVersions/Installer.php]: `chmod($installPath, 0664 & ~umask())`?

**Comment by Ocramius** [comment]
Merging as per @malukenho's approval

**Pull request merged**